apkid/rules/apk/protectors.yara

```diff
@@ -170,3 +170,19 @@ rule appdefence : protector
   condition:
     is_apk and all of them
 }
+
+rule dpt_shell : protector
+{
+  meta:
+    description = "DPT Shell"
+    url         = "https://github.com/luoyesiqiu/dpt-shell"
+    sample      = "0c4341700f4e685cafc9c86c9112098b75057580ba1f7163bc971347af3712ad"
+    author      = "Abhi"
+
+  strings:
+    $asset = "assets/app_name"
```

Comment:
The `asset/OoooooOooo` is expected to become random, which may render it less useful in the near future. The `app_name` file is currently empty, and the `app_acf` contains `androidx.core.app.CoreComponentFactory`. Perhaps we could implement a check for it, in case they decide to randomize that as well, ensuring the contents remain unchanged.

Reply:
will this do?
```yara
  strings:
    $asset = "assets/app_name"
    $asset_acf = "assets/app_acf"
    $assetlib = /assets\/(.*)\/(arm.*|x86.*)\/libdpt\.so/

  condition:
    is_apk and $assetlib and any of ($asset, $asset_acf)
```

Reply:
If even `libdpt.so` is present...hmm... yeah nice

Reply:
This one looks more better and reasonable imo:
`is_apk and ( $assetlib or any of ($asset, $asset_acf))`

Reply:
Looks like combination of both :haha:

Reply:
> Or even better to extend your first condition
> 
> `is_apk and $assetlib and any of ($asset*)`

This is better ig

Reply:
Wait then should i change the naming of `$assetlib` since it'll come in `any of ($asset*)`, NO?

Reply:
@enovella I believe it's all good now, do check